lib/livebook_web/live/settings_live.ex

```diff
@@ -286,16 +335,32 @@ defmodule LivebookWeb.SettingsLive do
     handle_event("set_autosave_path", %{}, socket)
   end
 
+  def handle_info({:env_vars_changed, env_vars}, socket) do
+    {:noreply, assign(socket, env_vars: env_vars)}
+  end
+
   def handle_info(_message, socket), do: {:noreply, socket}
 
   defp autosave_dir() do
-    Livebook.Settings.autosave_path()
+    path =
+      case Livebook.Settings.autosave_path() do
+        nil -> System.tmp_dir!()
```

Comment:
A better default would be `default_autosave_path`, but neither is really true. We currently only use `nil` for tests, for I think we can just match `autosave_path_select(%{state: %{file: nil}})` and render nothing :)

Reply:
The failure occurs here: `Livebook.FileSystem.Utils.ensure_dir_path/1`. So we will need to change how we can get this path, or maybe handle it from the function:

```elixir
    case storage().fetch_key(:settings, "global", :autosave_path) do
      {:ok, value} when is_binary(value) -> value
      _ -> default_autosave_path()
    end
```

WDYT?

Reply:
We still want to return `nil` from the settings, because that's what actually disables autosaving in tests :)